src/metadata-builder/ClosureJunctionEntityMetadataBuilder.ts

```diff
@@ -32,7 +32,7 @@ export class ClosureJunctionEntityMetadataBuilder {
             connection: this.connection,
             args: {
                 target: "",
-                name: parentClosureEntityMetadata.tableNameWithoutPrefix,
+                name: parentClosureEntityMetadata.treeOptions && parentClosureEntityMetadata.treeOptions.closureTableName ? parentClosureEntityMetadata.treeOptions.closureTableName : parentClosureEntityMetadata.tableNameWithoutPrefix,
```

Comment:
This and the column names should probably be extracted to a variable since they are very long.

Reply:
This might not have been the intended change here, see also https://github.com/typeorm/typeorm/blob/c4a36da62593469436b074873eba186f0f8b990d/src/metadata/EntityMetadata.ts#L792